packages/superset-ui-chart-controls/src/utils/advancedAnalytics/rollingWindow.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitationsxw
+ * under the License.
+ */
+import {
+  ensureIsArray,
+  ensureIsInt,
+  QueryFormData,
+  QueryObject,
+  RollingType,
+} from '@superset-ui/core';
+
+export function rollingWindowTransform(
+  formData: QueryFormData,
+  queryObject: QueryObject,
+): QueryObject {
+  if (
+    formData.rolling_type === RollingType.None ||
+    !formData.rolling_type ||
+    formData.metrics === undefined
+  ) {
+    return queryObject;
+  }
+
+  // ensure `post_processing` is a copy from queryObject
+  const post_processing = ensureIsArray(queryObject.post_processing).slice();
+  const columns = Object.fromEntries(
+    formData.metrics.map(metric => {
+      if (typeof metric === 'string') {
+        return [metric, metric];
+      }
+      return [metric.label, metric.label];
+    }),
+  );
+  if (formData.rolling_type === RollingType.Cumsum) {
+    // rolling must be the first operation(before pivot or other transforms)
+    post_processing.unshift({
+      operation: 'cum',
+      options: {
+        operator: 'sum',
+        columns,
+      },
+    });
+  }
+
+  if ([RollingType.Sum, RollingType.Mean, RollingType.Std].includes(formData.rolling_type)) {
```

Comment:
`else if` (no use in calling this if the `if` statement about is true)

Reply:
If these conditions are not matched. This function will not do any processing.

Reply:
`formData.rolling_type` can never be equal both to `RollingType.Cumsum` AND `RollingType.Sum`, which is why I was saying you should use an `else if` here.

Reply:
`RollingType.Cumsum` in the above processing, no need to use `else` at here. 

btw, I refactor `post processing` logic today. you can review the latest codes. thank you!

Reply:
Sorry, I see that now it's returning from the `if` statement whereas before it wasn't. Looks good now

Reply:
No biggie! Thanks for the review again.